syntax = "proto3";

enum SensorChannelType {
  SENSOR_TEMP = 0;
}

message SensorChannel {
 uint32 idx = 1;
 SensorChannelType type = 2;
}

message SensorData {
  uint64 timestamp = 1;
  SensorChannel channel = 2;
  float value = 3;
}

message PubSub {
  uint32 publisher_id = 1;
  oneof topic_msg {
    SensorData sensor_data = 2;
  }
}

